src/main/scala/org/apache/mesos/chronos/scheduler/state/MesosStatePersistenceStore.scala

```diff
@@ -59,11 +64,26 @@ class MesosStatePersistenceStore @Inject()(val zk: CuratorFramework,
     }
   }
 
+  def removeTaskFlowState(id: String): Boolean = {
+    remove(flowName(id))
+  }
+  
+  def getTaskFlowState(id: String): TaskFlowState = {
```

Comment:
Thanks @elingg for taking a look at my changes. 

You are correct, there are many ways to implement this. As you mentioned, an alternate way would be to change all the jobs to execute commands like: set_env && exec job where set_env would go out to a key value store. The question here becomes what is my primary key and what is responsible for updating the key value store. 

One option is to make the primary key the name of the job. However, this would make it impossible to ever run two instances of the same task flow simultaneously. Admittedly for cron jobs restarting the job while the previous one is still running is most likely a bad idea. However, for jobs that are triggered and parameterized it starts to make sense to allow multiple instances of a task flow to run simultaneously given there are resources available. (this is also a requirement for me)  

Alternatively a task flow id could be the primary key. This task id would have to be propagated and managed by chronos given that I do not want to replicate the scheduler logic elsewhere. This id would probably have to be an argument to the trigger command in order for the parameters to be properly set up before the job starts. While this works, I am not enthusiastic about it because it adds complexity to the requester and, by this point, it almost seems like I should be able to add parameters too.

I am all for allowing child jobs to inherit their parent environments. That is what I was trying to get at with this feature. However, I hit the 255 character limit on the mesos task id. Were you thinking of an alternate way of doing this? 

What is the hesitancy behind adding another layer of dependency on zookeeper? While zookeeper has its drawbacks, from my understanding, chronos will always need some sort of replicated store to work as a job scheduler. I don’t believe this addition will make it any harder to substitute in a different key/value store if that is desired. Additionally I don’t see parameters as being too big to store in zookeeper. What am I missing here? 

Also, for some perspective, it seems that another scheduler Azkaban supports "flow parameters”: http://azkaban.github.io/azkaban/docs/2.5. This doesn’t necessarily make it any any better or worse of an idea for chronos to support them too.